src/OpenTelemetry.Resources.AWS/AsyncHelper.cs

```diff
@@ -0,0 +1,44 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+namespace OpenTelemetry.Resources.AWS;
+
+/// <summary>
+/// A helper class for running asynchronous methods synchronously.
+/// </summary>
+internal static class AsyncHelper
+{
+    private static readonly TaskFactory TaskFactory = new TaskFactory(
+        CancellationToken.None,
+        TaskCreationOptions.None,
+        TaskContinuationOptions.None,
+        TaskScheduler.Default);
+
+    /// <summary>
+    /// Executes an async task which has a void return value synchronously.
+    /// </summary>
+    /// <param name="task">The async task to execute.</param>
+    public static void RunSync(Func<Task> task)
+    {
+        TaskFactory
+            .StartNew(() => task(), CancellationToken.None, TaskCreationOptions.None, TaskScheduler.Default)
```

Comment:
```suggestion
            .StartNew(() => task())
```

This code could be simplified in this way but in this case a warning CA2008 is raised because of https://github.com/dotnet/roslyn-analyzers/issues/1907. I can let the implementation as is or simplify it + pragma warning disable.

Reply:
Leaving it as is is fine.